### DISCUSSION TITLE
Welcome to cigale Discussions!

### DISCUSSION THREAD

**Discussion by mboquien** [mboquien/cigale#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by universpider** [comment]
Hello, I encountered a problem when I used the CIGALE to fit the galaxy sed.

First, we tested the example, the result with positive and negative residuals is normal. 

However, when we use our data to fit. The fit result is shown in this figure. 
[KUG0810_best_model.pdf](https://github.com/mboquien/cigale/files/7222932/KUG0810_best_model.pdf)

The reduced Chi^2 is small, however, the relative residual fluxes (Obs-Mod/Obs) are all positive. It seems unnatural, because we think the residuals should be positive and negative. We have tried several models(sfh2exp,sfhdelayed,dustatt_powerlaw,dustatt_modified_CF00,dale2014,dl2014), but the results are similar, all of the residuals are still positive. 

Thanks.

**Reply by mboquien** [reply]
That indeed should not happen. Could you attach the flux file and the pcigale.ini+pcigale.ini.spec please? I will look into it. Also, which version as you using? It reminds me of a very old bug (long corrected) where the best-fit was incorrect in some rare cases.

**Reply by Guang91** [reply]
They had sent me the files from e-mail previously.
But I got some error when running it using "master" branch. 
Maybe it's a version issue? 

[files.zip](https://github.com/mboquien/cigale/files/7226060/files.zip)

**Reply by mboquien** [reply]
Somehow I could not find this file. At any rate, with some small adaptations for the latest version, it seems to work fine for me.

![image](https://user-images.githubusercontent.com/444815/134685803-48aa1371-6b01-45b8-8b09-081522cdba64.png)

**Reply by Guang91** [reply]
I guess you refer to the latest "master" branch (aka official release one)?  or the "develop" branch?

**Reply by mboquien** [reply]
I am referring to the develop branch.

**Reply by universpider** [reply]
Thanks for the quick reply. You are right, it is the version issue. I downloaded the old version from http://cdsarc.u-strasbg.fr/viz-bin/qcat?J/A+A/622/A103. When I use the latest version, the result is fine. 
Thanks again for your time.

**Comment by mjrutkowski** [comment]
Hi,
I've used CIGALE fairly often (not a superuser but not a novice), but I'm curious about the validity of the results when I use in a new (to me) way to derive a new parameter of interest.   Specifically, when have photometry of high redshift galaxies which probe blueward of the rest-frame Lyman break regime. I'm curious as to whether I can trust the best fit nebular f_esc value that is provided with the best-fit model results using either of the attenuation laws (CF00 or Calzetti) which aren't (neither I believe) modeled in this regime.  Are the models extrapolated there? Here's a result using the built-in plot of a galaxy of interest for reference.  There was no AGN/dust contribution here,  with the data being limited to the optical. What say you? I think the distinction between the attenuated and best-fit seems a bit off here ... and my first thought is that attenuation itself causes an issue.  Thanks in advance.

<img width="1048" alt="Screen Shot 2021-09-24 at 2 57 47 PM" src="https://user-images.githubusercontent.com/5719235/134733127-04336971-95cf-411f-9bfc-3361258b6192.png">

**Reply by mboquien** [reply]
I would be very cautious about what is below the Lyman break regarding stars+gas with the current version. You are right that for now the curves are extrapolated, which is far from ideal. However, we are working on having them replaced with computed curves below the break. Attenuation curves are very uncertain in that regime but it would certainly much better. There is a first implementation for the skirtor AGN model. I have started to work on a patch for the attenuation modules but I have not found the time to finish it yet.

**Reply by mjrutkowski** [reply]
Thanks for such a quick reply! In some low-redshift work I was impressed that the fitted (with cigale) and measured (from spectra) f_esc matched as well as they did (still in prep.), but from this higher redshift work I'm beginning to second guess that.  
There's a broader issue here as well...which may be misinterpreting the process. 
Here I used two component stellar populations, both relatively young but only the youngest producing significant LyC (1Myr).  In the model production, the attenuated (here I used CF00) LyC starlight is reprocessed into the nebular emission, correct? Here, the best-fit f_esc in the model was 15% (f_esc was allowed to vary between 0-0.5, with a step of ~0.03-0.05). As such, I would expect then nebular lines to be significant (they are clearly) in the model, but the best-fit model is ~10% that of the attenuated component.  Naively, if 1-f_esc is reprocessed in the LyC, I wouldn't expect so dramatic a shift, right?  
 Does that sound reasonable/clear?  Perhaps I should prepare a document outlining this more specifically ?  I'd like to make sure I'm being safe in my implementation/interpretation of the code.  NO papers have been submitted or in prep., but with JWST ramping up of course we'd like to get HST papers out of the door to "prime the pump" so to speak.  I'd like to do my due diligence in the application of this VERY nice code (I find it quite useful, especially as I've moved on from IDL) early in the process you know? I'm happy to discuss this "off-line" too. Any thoughts are welcome. Cheers.

**Reply by mboquien** [reply]
I looked at the CF00 code. For now we cut the attenuation at 91.2 nm. The idea was that initially we were never fitting anything below this (rest-frame) and constraints on the escape fraction were through indirect means and it was expected to be small anyway, so only limited effect on the IR. Now with the addition of X-rays, it is becoming necessary to have a reasonable curve down to a few nm not to introduce nonphysical breaks in the spectra, so we have introduced an extension for the AGN. I still need to do that for the stellar populations. Regarding your question “In the model production, the attenuated (here I used CF00) LyC starlight is reprocessed into the nebular emission, correct?”, I am not certain what you mean exactly. The “escaped” fraction is partially absorbed by the dust in the same way as the stellar populations at the same wavelength. That is, it is the escape from gas, not from dust. Note that at higher z, there is also the added effect of the IGM, which can be very large and adds another layer of uncertainty. In any case, this is getting a little detailed, let's discuss this over email. Once I understand exactly what you need, I will move any necessary improvement to the top of my TODO list.

**Comment by alessandropeca** [comment]
Hi, I am new to Cicale and I am using the new January 2022 version. When I use the command `pcicale-plot` sed, I get the SED in the observed frame. Is there any way to get it in the rest-frame?
Thank you.

**Reply by mboquien** [reply]
Ciao Alessandro. You can use the `--type lum` option. You can get all the options with `pcigale-plots sed --help`.

**Reply by alessandropeca** [reply]
Ciao! Thank you so much, this `pcigale-plots sed --help` helps a lot.

**Comment by thriveth** [comment]
Hi, I am completely new to Cigale but so far it looks very useful. I am however having some trouble getting photometric redshift mode to work. I have set the redshift column of my objects to be -1 (the docs just say to make it negative, does not specify any number), and have then set `eval np.arange(0.1, 6., 0.1) in the redshift field in pcigale.ini. However, once the run is done, the best-fit SED comes out with a redshift of 1, no decimal places. This is both a poor fit for the specific galaxies in question, but also seems like an unlikely outcome of a best-fit solution, so I wonder what I am doing wrong here?

**Reply by mboquien** [reply]
Hello. This is indeed not the expected behavior. Could you send me a minimal test case (for instance a single object with the pcigale.ini and pcigale.ini.spec files) so I can look into the issue? Also, which version are you using? Thanks!

**Reply by thriveth** [reply]
Thank you! 
I am using the latest version, just installed and downloaded, with Python 3.9 in a newly created Conda environment. 

I have attached the files for an minimal case, renamed to .txt si Github would accept it. 

[Circ_apts_for_cigale_mJy_varz.txt](https://github.com/mboquien/cigale/files/7912203/Circ_apts_for_cigale_mJy_varz.txt)
[pcigale.txt](https://github.com/mboquien/cigale/files/7912204/pcigale.txt)

**Reply by thriveth** [reply]
Oops, forgot the .ini.spec file... here it is as .txt.txt

[pcigale.txt.txt](https://github.com/mboquien/cigale/files/7913496/pcigale.txt.txt)
.

**Reply by mboquien** [reply]
Sorry for the delay, I was in meetings most of the day. I have given a try with CIGALE 2022.0 (the very latest version released last week). I do not have most of your filters at hand so I tried with just the first few filters for the object `F336W_606`. It appears to work as expected. Both bayes.universe.redshift and best.universe.redshift seem OK and the fit is extremely good. I am not sure what may be happening here. Can you show the information panels displayed by `pcigale check`?

**Reply by thriveth** [reply]
Hi again, and thank you for your quick and thorough replies! 

Here's the output of `pcigale check`:

![image](https://user-images.githubusercontent.com/1553566/150608477-fdf95cff-03b1-44ac-9338-822e0aa22bb9.png)

If it is any help, I can zip a package of the filters (HST filters generated with `stsynphot` and saved as text files).

**Reply by thriveth** [reply]
By the way @mboquien Which results did it yield for the redshift when you ran it on F336W_606?

**Reply by mboquien** [reply]
Yes please, it would be useful to have the exact same filters. I did the test with just 4 filters, so I do not think the results should really be looked at. I will let you know what I get with all the filters.

**Comment by jonhbs** [comment]
Hello to everyone.

I'm going to start to work with CIGALE for the first time, so for the installation, I'm following the "How to" document. And I was wondering if: is it necessary to create a new Conda environment for installing CIGALE? If yes, why is it recommended? 

Thank you very much. Best regards

**Reply by mboquien** [reply]
It is not necessary. You can use an existing environment or not use Conda at all. Personally I never use Conda, but it boils down on personal preferences. Note however that the latest version of CIGALE requires Python 3.8 or more recent.

**Reply by Guang91** [reply]
If you can have python>=3.8 and install all the following packages 
astropy, numpy, scipy, sqlite, sqlalchemy, matplotlib, configobj, rich
then there's no need for a new environment. 

But if conflicts occur (e.g., your base env has python<3.7), the simplest solution is to create a new environment.

**Reply by jonhbs** [reply]
Understood... At the moment I have Python 3.9.7 installed.

Thank you very much for your soon reply and support. 

Best Regards from Colombia

**Comment by ritish142857** [comment]
I'm using the latest version of cigale2022 in order to fit the sed of two quasars. For one source I have got the good fitting but for another source, I am not able to get the fitting perhaps they are likely to be the same type of source i.e quasar. I have attached the pcigale .ini file and plot for both the source, if you have time please give some suggestions like how I can improve my fitting.
[pcigale _1st.ini.txt](https://github.com/mboquien/cigale/files/7948965/pcigale._1st.ini.txt)
[11111_best_model.pdf](https://github.com/mboquien/cigale/files/7948966/11111_best_model.pdf)
[pcigale _2nd.ini.txt](https://github.com/mboquien/cigale/files/7948967/pcigale._2nd.ini.txt)
[22222_best_model.pdf](https://github.com/mboquien/cigale/files/7948968/22222_best_model.pdf)

**Reply by mboquien** [reply]
What I would suggest is to use more than a single AGN model, in particular as the the spectrum is entirely AGN-dominated. Also, that would not change your results, but I would suggest to upgrade to CIGALE 2022.0. The different AGN components would be visible on the plots, which would help for the diagnostic.

**Reply by Guang91** [reply]
Another thing to consider is to try different AGN opening angle (oa). Maybe also viewing angle (i) to make sure the model is always type 1. 
oa can affect the ratio between UV/optical vs. IR.

**Comment by alessandropeca** [comment]
Hi,

I am using CIGALE in the "photometric mode". Is there any way to get the PDZ?

**Reply by mboquien** [reply]
You can set the `save_chi2` option to `properties` in the options of `pdf_analysis`. Note that this can take quite a bit of disk space. Once this done, you can run `pcigale-plots pdf` to get a quick look at the PDF.

**Reply by alessandropeca** [reply]
Thanks for the reply. That is what I tried without success. I noticed that CIGALE produces the PDFs under the options `variables` in the `[analysis_params]` section. Maybe should I add something about redshift here, the only option I see is `universe.redshift`, but it doesn't seem like what I need.

**Reply by alessandropeca** [reply]
Just to be clear, I would like to get the PDF of the photo-z.

**Reply by mboquien** [reply]
It should indeed be `universe.redshift`. Would you mind sending me your `pcigale.ini` so I have a quick look? Thanks!

**Reply by alessandropeca** [reply]
Sure, this is mine `pcigale.ini`
[pcigale.ini.txt](https://github.com/mboquien/cigale/files/8111222/pcigale.ini.txt)

From the last run I got this PDF:
[1094_universe.redshift_pdf.pdf](https://github.com/mboquien/cigale/files/8111234/1094_universe.redshift_pdf.pdf)

Which is "ok", but the best-fit redshift, which is printed on the `best_model.pdf` is `z=3.2`, which doesn't correspond to the peak in the PDF, which is around 2.4.

**Comment by alessandropeca** [comment]
Hi,

is there any command to parallelize `pcigale run`?
It would be very helpful when fitting huge catalogs!

**Reply by mboquien** [reply]
Ciao Alessandro. It is already parallel. You can change the number of cores in the pcigale.ini file. This being said when you have a huge catalog (how many are we talking about in this specific case?), the initial processing of the catalog is serial and can take quite a bit of time. Once done, the parallel processes for the computations of the models and the fitting will start automatically.

**Reply by alessandropeca** [reply]
Thank you for the answer! We are currently working on a catalog containing millions of sources.

**Comment by peteroffsapientia** [comment]
Hello to everyone.

I am a novice to CIGALE. I have installed the 2020 version and followed
[CIGALE-v0.9.0-HowTo.pdf](https://github.com/mboquien/cigale/files/8604320/CIGALE-v0.9.0-HowTo.pdf)
the CIGALE-v0.9.0-HowTo.pdf (here attached) for some practice.
Truth is; I don't understand what is happening since am a NOVICE. 
From the practice
1) how does one generate the Cigale_Test.mag file?
2)Is there a practical guide (for dummies) that explains how to use CIGALE?

@mboquien @Guang91 I am learning CIGALE to estimate the Star formation rate and the Attenuation curve of dwarf galaxies.
Thank you

**Reply by Guang91** [reply]
We have a new manual https://cigale.lam.fr/documentation/
It has practical instructions about how to generate the input and etc. 
Let me know if you have any further questions.

**Comment by Negar3362** [comment]
Hi. I have two questions regarding SED fitting with Cigale (version 2018).

1. I use Calzetti et al. 2000 to model attenuation and estimate attenuation for both young and old stellar populations for WISE-detected star forming galaxies. I am a bit confused which value should be utilized to correct optical line flux for extinction : E(B-V) of young stellar population or E(B-V)gas = E(B-V)star/0.44
To correct flux I use: F_int = F_abs 10^(0.4 E(B-V) kappa)

2. Why does the stellar library from Maraston 2005 (m2005) not work with a nebular model? 
Thanks.

**Reply by mboquien** [reply]
1. If you want a bona-fide Calzetti curve you should use the `dustatt_modified_starburst` module. The continuum attenuation is based on E(B-V)stars (https://gitlab.lam.fr/cigale/cigale/-/blob/develop/pcigale/sed_modules/dustatt_modified_starburst.py#L394).
2. In our tests we found that the Lyman continuum photon production was not what we would expect, so as to avoid possible issues on the reliability of the nebular emission with `m2005`, we disabled the combination.

**Reply by Negar3362** [reply]
Thanks for the reply.

**Comment by alessandropeca** [comment]
Hi,

What the option `-c CONFIG_FILE` actually works? Does it point to a pcigale.ini file that is not in the current folder? Or does it point to another file(s)?

For example, if I am in the main folder where there is a pcigale.ini file, but I want to use another one, can I just use `-c new_pcicale.ini`?

Thank you in advance.

**Reply by mboquien** [reply]
Yes, it works as you describe, you pass a configuration file. Remember to also have the `new_pcigale.ini.spec` file in the same directory as `new_pcigale.ini`.

**Comment by danielschaerer** [comment]
Hi Mederic,
is it possible to use BPASS models with CIGALE ? 
In any case, thanks for the great code.
Daniel

**Reply by mboquien** [reply]
Yes. You need to either use the `git` version and switch to the `bpass` branch or download the [latest snapshot](https://gitlab.lam.fr/cigale/cigale/-/archive/bpass/cigale-bpass.tar.gz). Then you need to download separately the bpass v2.2.1 models and copy all the files starting with `spectra`, `ionzing` and `starmass` directly into the `database_builder/bpassv2.2/` folder. Then you can install CIGALE as usual. The name of the corresponding module is `bpassv2`. I hope it helps.

**Reply by danielschaerer** [reply]
Thanks, will try ! 

Then, has the nebular emission also been computed with these same BPASS SEDs ? I guess not, since would need many additional CLOUDY models.

**Reply by mboquien** [reply]
Not yet. It is on our TODO list.

**Comment by danielschaerer** [comment]
Hi again !
The new "nebular" module allows for variations of the nebular metallicity 'zgas' which are a priori independent of the stellar metallicity. 

Does it mean that CIGALE will compute all the combinations of stellar and gas metallicity ? (including some which may not make physical sense)

How does one constrain the code to us e.g. only the combinations of 'zgas' and 'metallicity' which were actually used to compute the CLOUDY modells (those in Table 4 of Villa-Vilez+2021, https://doi.org/10.1051/0004-6361/202140890 )?

**Reply by mboquien** [reply]
If you include several values of both Zgas and Zstars, it will indeed compute all the combinations. However CIGALE does not know whether this is physical or not (for instance the accretion of pristine gas can lead to Zgas<Zstar).

As for your second question, as far as I can tell, with the latest version of CIGALE, the nebular models do not seem to be directly dependent on the stellar metallicity. In any case what you can do is to have several runs with just one or few metallicities that are appropriate for your case.

**Comment by ghost** [comment]
Hi! 
I am completely new to CIGALE and was trying to get the SED for certain sources from HSC and VISTA. But I am not getting a proper FIT for the IR data. I don't know why that is happening. I tried adjusting the parameters for the modules but not able to see a large difference. Could anybody help me out with the issue? (I have attached the configuration file and the input file. 
[pcigale-new.txt](https://github.com/mboquien/cigale/files/13850070/pcigale-new.txt)
[HSCIRcatalog.txt](https://github.com/mboquien/cigale/files/13850071/HSCIRcatalog.txt)
[J000913.63-000112.6_best_model (14).pdf](https://github.com/mboquien/cigale/files/13850072/J000913.63-000112.6_best_model.14.pdf)
( Could it by any change be the issue with the flux values?)

**Reply by mboquien** [reply]
See answer in the ticket you opened.
